doc/ipythonconsole.rst

```diff
@@ -2,91 +2,159 @@
+For this feature to work, a compatible version of the ``spyder-kernels`` package :ref:`must be installed <starting-kernel-problems-ref>` in the environment or machine in which the external kernel is running.
 
 .. image:: images/console/console-menu.png
-   :alt: Spyder IPython Console as above, with the options menu open
+   :alt: Connect to external kernel dialog of the Spyder IPython console
 
 
-Connect to an external kernel
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Connect to a local kernel
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. note::
-   If on Windows and connecting to a remote machine over ``ssh``, you'll need to install the ``paramiko`` python package first, *e.g.* with ``conda install paramiko`` if using Anaconda.
+To connect to a local kernel that is already running (e.g. one started by Jupyter notebook),
 
-To connect to an external kernel,
+#. Run ``%connect_info`` in the notebook or console you want to connect to, and copy the name of its kernel connection file, shown after ``jupyter <app> --existing``.
 
-#. Launch an IPython kernel on the local or remote host if one is not already running.
+   .. image:: images/console/console-connect-local-step1.gif
+      :alt: Running connect_info in a Jupyter notebook
 
-   * If using Spyder 3.3.0 or later, you'll need to launch the kernel with ``python -m spyder_kernels.console`` (after you've first installed ``spyder-kernels`` on the host with ``<conda/pip> install spyder-kernels``).
-   * If using a version of Spyder before 3.3.0, ``ipython kernel`` should work to launch the kernel, albeit without certain Spyder-specific features.
+#. In Spyder, click :guilabel:`Connect to an existing kernel` from the :guilabel:`Consoles` menu, and paste the name of the :guilabel:`Connection file` from the previous step.
 
-#. Copy the connection file (:file:`{jupyter/runtime/dir/path}/kernel-{pid}.json`) to the machine you're running Spyder on (if remote) or note its location (if local).
+   As a convenience, kernel ID numbers (e.g. ``1234``) entered in the connection file path field will be expanded to the full path of the file, i.e. :file:`{jupyter/runtime/dir/path}/kernal-{id}.json`.
 
-   You can get :file:`{jupyter/runtime/dir/path}` by executing ``jupyter --runtime-dir`` in the same Python environment as the kernel.
+   .. image:: images/console/console-connect-local-step2.gif
+      :alt: Copying the connection filename into Spyder's dialog
 
-#. Click :guilabel:`Connect to an existing kernel` from the :guilabel:`Console` menu or the :guilabel:`IPython Console` pane's "Gear" menu.
+#. Click :guilabel:`OK` to connect to the kernel.
 
-#. Browse for or enter the path to the connection file from the previous step.
-   If you're connecting to a local kernel, click :guilabel:`Ok` and Spyder should connect to the kernel; if a remote kernel, proceed to the final step.
+   .. image:: images/console/console-connect-local-step3.gif
```

Comment:
I suggest to only print `test_var` in this gif. The first `print` seems unnecessary.

Also, if you can use a smaller screen resolution would be nice to see if you can avoid the panning you need to do to go to the console.

Reply:
Both good points, it just will require some time to get everything set up here to redo everything, so given family commitments they probably won't be ready until tomorrow (assuming I can get everything to work).

Reply:
Redone to fix